package-res/resources/web/pentaho/lang/SortedList.js

```diff
@@ -93,11 +95,24 @@ define([
     },
 
     set compare(compareFunction) {
-      this._compareFunction = compareFunction;
+      var hasOwn = O.hasOwn(this, '_compareFunction');
+      if (hasOwn && this._compareFunction !== compareFunction || !hasOwn && compareFunction != null) {
+        if (compareFunction == null) {
+          O.delete(this, '_compareFunction');
```

Comment:
I don't think that there is a reason to use the `O.delete` method here. Actually, I think it's not being used anymore and we should probably remove it.